src/common/diffHunk.ts

```diff
@@ -262,61 +251,25 @@ export function getGitChangeType(status: string): GitChangeType {
 	}
 }
 
-export async function parseDiff(reviews: any[], repository: Repository, parentCommit: string): Promise<(RichFileChange | SlimFileChange)[]> {
-	let fileChanges: (RichFileChange | SlimFileChange)[] = [];
+export async function parseDiff(reviews: any[], repository: Repository, parentCommit: string): Promise<(InMemFileChange | SlimFileChange)[]> {
+	let fileChanges: (InMemFileChange | SlimFileChange)[] = [];
+
 	for (let i = 0; i < reviews.length; i++) {
 		let review = reviews[i];
+
 		if (!review.patch) {
 			const gitChangeType = getGitChangeType(review.status);
 			fileChanges.push(new SlimFileChange(review.blob_url, gitChangeType, review.filename));
 			continue;
 		}
 
-		if (review.status === 'modified') {
-			let fileName = review.filename;
-			try {
-				let originalContent = await getFileContent(repository.path, parentCommit, fileName);
-				let richFileChange = await parseModifiedHunkComplete(originalContent, review.patch, fileName, fileName);
-				richFileChange.blobUrl = review.blob_url;
-				fileChanges.push(richFileChange);
-			} catch (e) {
-				let richFileChange = await parseModifiedHunkFast(review.patch, fileName, fileName);
-				richFileChange.blobUrl = review.blob_url;
-				fileChanges.push(richFileChange);
-			}
-		} else if (review.status === 'removed' || review.status === 'added' || review.status === 'renamed') {
-			if (!review.patch) {
-				continue;
-			}
+		const gitChangeType = getGitChangeType(review.status);
 
-			const gitChangeType = getGitChangeType(review.status);
-			let contentArray = [];
-			let fileName = review.filename;
-			let prDiffReader = parseDiffHunk(review.patch);
-			let prDiffIter = prDiffReader.next();
-			let diffHunks = [];
-
-			while (!prDiffIter.done) {
-				let diffHunk = prDiffIter.value;
-				diffHunks.push(diffHunk);
-				for (let j = 0, len = diffHunk.diffLines.length; j < len; j++) {
-					let diffLine = diffHunk.diffLines[j];
-					if (diffLine.type !== DiffChangeType.Control) {
-						contentArray.push(diffLine.text);
-					}
-				}
-				prDiffIter = prDiffReader.next();
-			}
-
-			let contentFilePath = await writeTmpFile(contentArray.join('\n'), path.extname(fileName));
-			let emptyContentFilePath = await writeTmpFile('', path.extname(fileName));
-			let richFileChange = review.status === 'removed' ?
-				new RichFileChange(emptyContentFilePath, contentFilePath, gitChangeType, fileName, diffHunks, false) :
-				new RichFileChange(contentFilePath, emptyContentFilePath, gitChangeType, fileName, diffHunks, false);
-			richFileChange.blobUrl = review.blob_url;
-			fileChanges.push(richFileChange);
-		}
+		let originalFileExist = await repository.checkFileExistence(parentCommit, review.filename);
```

Comment:
I think we need to be able to distinguish files changes that are additions. This sets the `isPartial` flag to true for file additions so the warning gets shown for them

Reply:
Good catch.